app/Http/Controllers/UserController.php

```diff
@@ -14,9 +29,53 @@ class UserController extends Controller
     public function index()
     {
         if (!isset($listUsers)) {
-            $listUsers = User::all();
+            $listUsers = $this->user->paginate(10);
         }
 
         return view('users.index')->with('listUsers', $listUsers);
     }
+
+    /**
+     * Show the form for creating a new resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function create()
+    {
+        return view('users.create');
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param \Illuminate\Http\Request $request request store data user
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function store(UserPostRequest $request)
+    {
+        $user = new User;
+        $user->full_name = $request->get('full_name');
+        $user->email = $request->get('email');
+        $user->password = bcrypt($request->get('password'));
```

Comment:
you shoul use event of model.https://laravel.com/docs/5.4/eloquent#events (use observer)